README.md

```diff
@@ -1,2 +1,206 @@
-# Fluent
-A brand new framework to build amazing android applications
+
+# Fluent 
+
+Fluent is a lightweight framework thats helps you to create an entirely Android Application following SOLID pattern with an unidirectional flow reactive or not.
+
+The main concepts in this architecture are : `State`, `View`, `Store`, `Job` and `Hub` 
+
+---
+>![fluent diagram](https://image.ibb.co/cAXPhH/Screen_Shot_2018_04_11_at_12_09_41.png)
+---
+
+## Concept Explanation
+
+### State
+
+We recommend to define one `State` for each one of your screens.
+
+It holds just `type() : StateType` and can be inherit to fit your needs.
+
+### StateType
+
+We already have some default kind of states, you can create your own type inheriting this class.
+The default states are: `Initial`, `Loading`, `Success` and `Error`
+
+### View
+
+That's the reference to your screen itself, you should create your kind of `View` that has specific `State` attached.
+
+Your `Activity` or `Fragment` needs to implement `View`
+
+```kotlin
+class LoginActivity : AppCompatActivity(), LoginView {
+    override fun bind(newState: LoginState) { ... }
+}
+```
+
+### Store
+
+It receives and manages all the possible states of a `View` so you don't need to worry, because we take care about statefulness and performance.
+
+
+### Job
+
+A job is a little piece of work which might look like an atomic operation. It's the only place where you can generate new states and push them to the `Store`.
+
+### Hub
+
+The `Hub` is maybe the most important and disruptive layer of this framework. It is responsible to `connect()` the `View` and the `Job`'s layers.
+
+You can combine and/or filter actions before perform the `Job` itself. It acts kind like a bridge binding user actions with the use cases. 
+
+We have already implemented the `disconnect()` method to release resources by disposing all the bound jobs.
```

Comment:
This last sentence is only true to the reactive module since the basic one has no Hub implementation. We should only explain to our users why this method exists.